cppcheck hackathon echo 65

MialLewis opened the pull request:
### Description of work
Fixes a few cppcheck defects by:
- making things const
- passing by reference
- returning const reference

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

#### Further detail of work
<!-- Please provide a more detailed description of the work that has been undertaken.
-->

### To test:

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

MialLewis committed:
remove cppcheck suppression passedByValue AlgorithmProgressDialogPresenter

MialLewis committed:
remove cppcheck suppressions 1-4

MialLewis committed:
remove cppcheck suppression 5-6

MialLewis committed:
remove cppcheck suppression 7-15

MialLewis committed:
correct suppression line and add unsaved file

MialLewis committed:
fix new cppsupressions

sf1919 commented:
@MialLewis Cancelling builds as looks to be a genuine failure on the Liunux system tests

MialLewis committed:
add uncommitted code

thomashampson committed:
Merge branch 'main' into cpp_check_hackathon_echo_65

thomashampson commented on buildconfig/CMake/CppCheck_Suppressions.txt.in:
```diff
@@ -719,7 +703,6 @@ constVariablePointer:${CMAKE_SOURCE_DIR}/qt/widgets/common/src/AlgorithmDialog.c
 constVariablePointer:${CMAKE_SOURCE_DIR}/qt/widgets/common/src/AlgorithmDialog.cpp:902
 constVariablePointer:${CMAKE_SOURCE_DIR}/qt/widgets/common/src/AlgorithmDialog.cpp:904
-passedByValue:${CMAKE_SOURCE_DIR}/qt/widgets/common/src/AlgorithmProgress/AlgorithmProgressDialogPresenter.cpp:62
 passedByValue:${CMAKE_SOURCE_DIR}/qt/widgets/common/src/AlgorithmProgress/AlgorithmProgressPresenter.cpp:44
```
```suggestion
```

thomashampson committed:
Remove unmatched cppcheck suppression

thomashampson commented on buildconfig/CMake/CppCheck_Suppressions.txt.in:
```diff
@@ -661,22 +661,6 @@ returnByReference:${CMAKE_SOURCE_DIR}/qt/scientific_interfaces/ISISReflectometry
-returnByReference:${CMAKE_SOURCE_DIR}/qt/widgets/common/inc/MantidQtWidgets/Common/FileFinderWidget.h:130
-returnByReference:${CMAKE_SOURCE_DIR}/qt/widgets/common/inc/MantidQtWidgets/Common/FileFinderWidget.h:142
-returnByReference:${CMAKE_SOURCE_DIR}/qt/widgets/common/inc/MantidQtWidgets/Common/FileFinderWidget.h:148
 returnByReference:${CMAKE_SOURCE_DIR}/qt/widgets/common/inc/MantidQtWidgets/Common/FitDomain.h:35
```
```suggestion
virtualCallInConstructor:${CMAKE_SOURCE_DIR}/qt/widgets/common/inc/MantidQtWidgets/Common/AlgorithmProgress/AlgorithmProgressDialogPresenter.h:34
returnByReference:${CMAKE_SOURCE_DIR}/qt/widgets/common/inc/MantidQtWidgets/Common/FitDomain.h:35
```

thomashampson committed:
Suppress new cppcheck defect

Cppcheck found a new non-trivial defect.
Add it to the suppressions list for later.

yusufjimoh reviewed (approved):
The changes seem fine to me
